source/dem/velocity_verlet_integrator.cc

```diff
@@ -130,5 +130,113 @@ VelocityVerletIntegrator<dim>::integrate(
     }
 }
 
+template <int dim>
+void
+VelocityVerletIntegrator<dim>::integrate(
+  Particles::ParticleHandler<dim> &particle_handler,
+  const Tensor<1, 3> &             g,
+  const double                     dt,
+  std::vector<Tensor<1, 3>> &      torque,
+  std::vector<Tensor<1, 3>> &      force,
+  const std::vector<double> &      MOI,
+  const typename DEM::dem_data_structures<dim>::cell_set
+    &mobile_cells_container)
+{
+  Point<3>           particle_position;
```

Comment:
I think you have the wrong logic here.
Instead of doing
loop over particles ->. get cell around particle. Which is super expensive
You would be better off doing
loop over cells -> if cell has correct status  get the particles therein -> integrate them

Reply:
Yeah, I didn't want to changed the structure here, but I know we talked about it. I will refactor that!

Reply:
It's so much easier and better actually